packages/headless/src/api/knowledge/knowledge-api-answer.ts

```diff
@@ -0,0 +1,177 @@
+  const message = JSON.parse(event.data);
+  const payload = JSON.parse(message.payload);
+  if (message.payloadType === 'genqa.headerMessageType') {
+    draft.answerStyle = payload.answerStyle;
+    draft.contentFormat = payload.contentFormat;
+    draft.isStreaming = true;
+    draft.isLoading = false;
+  }
+
+  if (message.payloadType === 'genqa.messageType') {
+    if (draft.answer === undefined) {
+      draft.answer = payload.textDelta;
+    } else {
+      draft.answer = draft.answer?.concat(payload.textDelta);
+    }
+  }
+
+  if (message.payloadType === 'genqa.citationsType') {
+    draft.citations = payload.citations;
+  }
+
+  if (message.payloadType === 'genqa.endOfStreamType') {
+    draft.generated = Boolean(payload.answerGenerated);
+    draft.isStreaming = false;
+  }
+};
+
+const onOpenStream = async (response: Response) => {
+  if (
+    response.ok &&
+    response.headers.get('content-type')?.includes('text/event-stream')
+  ) {
+    return;
+  }
+
+  const isClientSideError =
+    response.status >= 400 && response.status < 500 && response.status !== 429;
+
+  if (isClientSideError) {
+    throw new FatalError({
+      message: 'Error opening stream',
+      code: response.status,
+    });
+  } else {
+    throw new Error();
+  }
+};
+
+const onError = (err: Error) => {
+  let retryCount = 0;
+
+  if (err instanceof FatalError) {
+    throw err;
+  }
+
+  if (++retryCount > 3) {
+    const error = {
+      message: 'Failed to complete stream.',
+      code: 1,
+    };
+    throw new FatalError(error);
+  }
+};
+
+export const answerAPI = knowledgeApi.injectEndpoints({
+  overrideExisting: true,
+  endpoints: (builder) => ({
+    getAnswer: builder.query<GeneratedAnswerStream, Partial<SearchRequest>>({
+      queryFn: () => ({
+        data: {
+          answerStyle: undefined,
+          contentFormat: undefined,
+          answer: undefined,
+          citations: undefined,
+          generated: false,
+          isStreaming: true,
+          isLoading: true,
+        },
+      }),
+      async onCacheEntryAdded(
+        arg,
+        {getState, cacheDataLoaded, updateCachedData}
+      ) {
+        await cacheDataLoaded;
+        /**
+         * createApi has to be called prior to create the redux store and is used as part of the store setup sequence.
+         * It cannot use the inferred state used by Redux, thus the casting.
+         * https://redux-toolkit.js.org/rtk-query/usage-with-typescript#typing-dispatch-and-getstate
+         */
+        const {configuration} =
+          getState() as unknown as StateNeededByKnowledgeAPI;
+
+        await fetchEventSource(
+          `${configuration.platformUrl}/rest/organizations/${configuration.organizationId}/answer/v1/configs/${configuration.knowledge.configurationId}/generate`,
+          {
+            method: 'POST',
+            body: JSON.stringify(arg),
+            headers: {
+              Authorization: `Bearer ${configuration.accessToken}`,
```

Comment:
how come you're not reusing the definition from the base query ?

Reply:
because the base query is not used by a `queryFn` 

I could have omited completely the `baseQuery` implementation here and the actual `answerApi` endpoint implementation would have worked exactly the same.